python/lsst/sconsUtils/builders.py

```diff
@@ -339,8 +339,9 @@ def __call__(self, env, config):
                           action="doxygen %s" % pipes.quote(outConfigNode.abspath))
         for path in self.outputPaths:
             env.Clean(doc, path)
-        env.Depends(doc, config)
-        self.results.extend(config)
+        if os.path.exists(config):
+            env.Depends(doc, config)
+            self.results.extend(config)
```

Comment:
I'm trying to understand this change for DM-35793, and I've managed to get very confused.

At first glance, this looks like a shadowing problem (`config` refers to two different objects over the lifetime of the function), but on second thought the original code makes a lot more sense.

At this point, `config` is a `Command` object that, if the `doxygen.conf.in` file is missing, has a target (presumably `doxygen.conf`?) but no  source. So it seems strange that calling `env.Depends` on this object would introduce a dependency on `doxygen.conf.in` when the object appears to have been constructed to not know about it. Am I missing something?